### PR TITLE
Feature/sc 35762

### PR DESCRIPTION
**Pull request by KuphJr**
## Changes

- Added better case matching in the Overrider for CoinGecko and CoinPaprika.  Now the case symbols requested by the job spec do not have to match the case of the symbols returned by the data provider.
- This should avoid errors in the future caused by mismatched casings in job specs

## Steps to Test

`yarn test coingecko & yarn test coinpaprika`

## Quality Assurance

- [x] Ran `yarn changeset` if adapter source code was changed
- [x] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [x] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [x] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)

### CODE REVIEW

File: packages/k6/limits.env [hunk 1]
```diff
@@ -1,8 +1,7 @@
...
-CI_ADAPTER_NAME=coingecko
-LOCAL_ADAPTER_NAME=
+LOCAL_ADAPTER_NAME=coinpaprika
 PAYLOAD_PATH=/config/http.json
-RPS=1000
-T=2
-UNIQUE_REQUESTS=100
+RPS=2
+T=10
+UNIQUE_REQUESTS=20
```
Comment (austinborn) [inline packages/k6/limits.env:7]: Are these relevant to the rest of the PR changes?

File: packages/k6/src/test-limits.ts [hunk 2]
```diff
@@ -98,7 +98,7 @@ export const options = {
     ],
   },
   stages: [
-    { duration: '5m', target: Math.min(uniqueRequests, VU / 10) }, // 5m warmup from 0 to min(uniqueRequests, VU/10)
+    { duration: '1m', target: Math.min(uniqueRequests, VU / 10) }, // 5m warmup from 0 to min(uniqueRequests, VU/10)
```
Comment (austinborn) [inline packages/k6/src/test-limits.ts:101]: Is this relevant to the rest of the changes?

File: packages/core/bootstrap/src/lib/modules/overrider.ts [hunk 3]
```diff
@@ -52,23 +52,24 @@ export class Overrider {
...
-      const symbol = coinResponse.symbol
+      const symbolUpperCase = coinResponse.symbol.toUpperCase()
+      const symbolLowerCase = coinResponse.symbol.toLowerCase()
 
-      if (remainingSyms.includes(symbol)) {
-        if (isOverridden[symbol] === true) {
-          if (!alreadyWarned[symbol])
-            logger.warn(
-              `Overrider: The symbol "${coinResponse.symbol}" has a duplicate coin id and no override.`,
+      if (remainingSyms.includes(symbolUpperCase) || remainingSyms.includes(symbolLowerCase)) {
```
Comment (austinborn) [inline packages/core/bootstrap/src/lib/modules/overrider.ts:58]: I'm worried about the run-time characteristics for this if the list of coins reaches into the hundreds. I'd recommend making `remainingSyms` into a map using the uppercase symbols, then turning the `includes` checks into `if (remainingSymsMap[symbolUpperCase])`.

File: packages/core/bootstrap/src/lib/modules/overrider.ts [hunk 4]
```diff
@@ -52,23 +52,24 @@ export class Overrider {
     const isOverridden: { [symbol: string]: boolean } = {}
     const alreadyWarned: { [symbol: string]: boolean } = {}
     for (const coinResponse of coinsResponse) {
-      const symbol = coinResponse.symbol
+      const symbolUpperCase = coinResponse.symbol.toUpperCase()
+      const symbolLowerCase = coinResponse.symbol.toLowerCase()
```
Comment (austinborn) [inline packages/core/bootstrap/src/lib/modules/overrider.ts:56]: Is it important to check uppercase and lowercase separately? At this point can we just make this completely case-agnostic?

**Review by austinborn — COMMENTED** [review]
The idea makes sense, although I had some questions about the implementation (explained in the comments).

**Comment by tateexon** [comment]
Soak test for coinpaprika failed: `
running (10h59m26.9s), 1/1 VUs, 4128 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m26.9s/12h0m0s

running (10h59m27.9s), 1/1 VUs, 4128 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m27.9s/12h0m0s

running (10h59m28.9s), 1/1 VUs, 4128 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m28.9s/12h0m0s

running (10h59m29.9s), 1/1 VUs, 4128 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m29.9s/12h0m0s

running (10h59m30.9s), 1/1 VUs, 4128 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m30.9s/12h0m0s

running (10h59m31.9s), 1/1 VUs, 4128 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m31.9s/12h0m0s

running (10h59m32.9s), 1/1 VUs, 4128 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m32.9s/12h0m0s

running (10h59m33.9s), 1/1 VUs, 4128 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m33.9s/12h0m0s

running (10h59m34.9s), 1/1 VUs, 4128 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m34.9s/12h0m0s

running (10h59m35.9s), 1/1 VUs, 4128 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m35.9s/12h0m0s

running (10h59m36.9s), 1/1 VUs, 4129 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m36.9s/12h0m0s

running (10h59m37.9s), 1/1 VUs, 4129 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m37.9s/12h0m0s

running (10h59m38.9s), 1/1 VUs, 4129 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m38.9s/12h0m0s

running (10h59m39.9s), 1/1 VUs, 4129 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m39.9s/12h0m0s

running (10h59m40.9s), 1/1 VUs, 4129 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m40.9s/12h0m0s

running (10h59m41.9s), 1/1 VUs, 4129 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m41.9s/12h0m0s

running (10h59m42.9s), 1/1 VUs, 4129 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m42.9s/12h0m0s

running (10h59m43.9s), 1/1 VUs, 4129 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m43.9s/12h0m0s

running (10h59m44.9s), 1/1 VUs, 4129 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m44.9s/12h0m0s

running (10h59m45.9s), 1/1 VUs, 4130 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m45.9s/12h0m0s

running (10h59m46.9s), 1/1 VUs, 4130 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m46.9s/12h0m0s

running (10h59m47.9s), 1/1 VUs, 4130 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m47.9s/12h0m0s

running (10h59m48.9s), 1/1 VUs, 4130 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m48.9s/12h0m0s

running (10h59m49.9s), 1/1 VUs, 4130 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m49.9s/12h0m0s

running (10h59m50.9s), 1/1 VUs, 4130 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m50.9s/12h0m0s

running (10h59m51.9s), 1/1 VUs, 4130 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m51.9s/12h0m0s

running (10h59m52.9s), 1/1 VUs, 4130 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m52.9s/12h0m0s

running (10h59m53.9s), 1/1 VUs, 4130 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m53.9s/12h0m0s

running (10h59m54.9s), 1/1 VUs, 4130 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m54.9s/12h0m0s

running (10h59m55.9s), 1/1 VUs, 4130 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m55.9s/12h0m0s

running (10h59m56.9s), 1/1 VUs, 4131 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m56.9s/12h0m0s

running (10h59m57.9s), 1/1 VUs, 4131 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m57.9s/12h0m0s

running (10h59m58.9s), 1/1 VUs, 4131 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m58.9s/12h0m0s

running (10h59m59.9s), 1/1 VUs, 4131 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  10h59m59.9s/12h0m0s

running (11h00m00.9s), 1/1 VUs, 4131 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  11h00m00.9s/12h0m0s

running (11h00m01.9s), 1/1 VUs, 4131 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  11h00m01.9s/12h0m0s

running (11h00m02.9s), 1/1 VUs, 4131 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  11h00m02.9s/12h0m0s

running (11h00m03.9s), 1/1 VUs, 4131 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  11h00m03.9s/12h0m0s

running (11h00m04.9s), 1/1 VUs, 4131 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  11h00m04.9s/12h0m0s

running (11h00m05.9s), 1/1 VUs, 4131 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  11h00m05.9s/12h0m0s

running (11h00m06.9s), 1/1 VUs, 4132 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  11h00m06.9s/12h0m0s

running (11h00m07.9s), 1/1 VUs, 4132 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  11h00m07.9s/12h0m0s

running (11h00m08.9s), 1/1 VUs, 4132 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  11h00m08.9s/12h0m0s

running (11h00m09.9s), 1/1 VUs, 4132 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  11h00m09.9s/12h0m0s

running (11h00m10.9s), 1/1 VUs, 4132 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  11h00m10.9s/12h0m0s

running (11h00m11.9s), 1/1 VUs, 4132 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  11h00m11.9s/12h0m0s

running (11h00m12.9s), 1/1 VUs, 4132 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  11h00m12.9s/12h0m0s

running (11h00m13.9s), 1/1 VUs, 4132 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  11h00m13.9s/12h0m0s

running (11h00m14.9s), 1/1 VUs, 4132 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  11h00m14.9s/12h0m0s

running (11h00m15.9s), 1/1 VUs, 4133 complete and 0 interrupted iterations
default   [  92% ] 1 VUs  11h00m15.9s/12h0m0s`

**Comment by tateexon** [comment]
Soak test for coingecko failed: `     ✗ Group-0-coingecko-LDO / ETH returned 200
      ↳  1% — ✓ 6 / ✗ 577
     ✗ Group-0-coingecko-PERP / USD returned 200
      ↳  0% — ✓ 4 / ✗ 579
     ✗ Group-0-coingecko-FOX / USD returned 200
      ↳  1% — ✓ 6 / ✗ 577
     ✗ Group-0-coingecko-IOST / USD returned 200
      ↳  2% — ✓ 12 / ✗ 571
     ✗ Group-0-coingecko-ALCX / ETH returned 200
      ↳  1% — ✓ 11 / ✗ 572
     ✗ Group-0-coingecko-RGT / ETH returned 200
      ↳  1% — ✓ 8 / ✗ 575
     ✗ Group-0-coingecko-ALBT / USD returned 200
      ↳  1% — ✓ 7 / ✗ 576
     ✗ Group-0-coingecko-LTC / USD returned 200
      ↳  1% — ✓ 7 / ✗ 576
     ✗ Group-0-coingecko-GTC / ETH returned 200
      ↳  1% — ✓ 6 / ✗ 577
     ✗ Group-0-coingecko-DASH / USD returned 200
      ↳  1% — ✓ 6 / ✗ 577
     ✗ Group-0-coingecko-UMEE / ETH returned 200
      ↳  2% — ✓ 14 / ✗ 569
     ✗ Group-0-coingecko-FIL / USD returned 200
      ↳  0% — ✓ 3 / ✗ 580
     ✗ Group-0-coingecko-LUNA / ETH returned 200
      ↳  2% — ✓ 15 / ✗ 568
     ✗ Group-0-coingecko-PLA / USD returned 200
      ↳  1% — ✓ 6 / ✗ 577
     ✗ Group-0-coingecko-STAKE / ETH returned 200
      ↳  0% — ✓ 5 / ✗ 578
     ✗ Group-0-coingecko-FEI / USD returned 200
      ↳  2% — ✓ 12 / ✗ 571
     ✗ Group-0-coingecko-CEL / ETH returned 200
      ↳  0% — ✓ 5 / ✗ 578
     ✗ Group-0-coingecko-TRX / USD returned 200
      ↳  0% — ✓ 1 / ✗ 582
     ✗ Group-0-coingecko-FTT / ETH returned 200
      ↳  1% — ✓ 11 / ✗ 572
     ✗ Group-0-coingecko-YFI / USD returned 200
      ↳  1% — ✓ 10 / ✗ 573
     ✗ Group-0-coingecko-KSM / USD returned 200
      ↳  1% — ✓ 7 / ✗ 576
     ✗ Group-0-coingecko-HBAR / USD returned 200
      ↳  2% — ✓ 15 / ✗ 568
     ✗ Group-0-coingecko-NMR / ETH returned 200
      ↳  1% — ✓ 6 / ✗ 577
     ✗ Group-0-coingecko-GUSD / USD returned 200
      ↳  0% — ✓ 1 / ✗ 582
     ✗ Group-0-coingecko-GRT / USD returned 200
      ↳  0% — ✓ 4 / ✗ 579
     ✗ Group-0-coingecko-YFI / ETH returned 200
      ↳  1% — ✓ 6 / ✗ 577
     ✗ Group-0-coingecko-AVAX / USD returned 200
      ↳  1% — ✓ 7 / ✗ 576
     ✗ Group-0-coingecko-HT / USD returned 200
      ↳  0% — ✓ 5 / ✗ 578
     ✗ Group-0-coingecko-SRM / ETH returned 200
      ↳  1% — ✓ 11 / ✗ 572
     ✗ Group-0-coingecko-STETH / USD returned 200
      ↳  0% — ✓ 5 / ✗ 578
     ✗ Group-0-coingecko-BNT / ETH returned 200
      ↳  1% — ✓ 9 / ✗ 574
     ✗ Group-0-coingecko-BETA / ETH returned 200
      ↳  0% — ✓ 5 / ✗ 578
     ✗ Group-0-coingecko-OCEAN / ETH returned 200
      ↳  1% — ✓ 9 / ✗ 574
     ✗ Group-0-coingecko-VGX / ETH returned 200
      ↳  0% — ✓ 4 / ✗ 579
     ✗ Group-0-coingecko-PAXG / ETH returned 200
      ↳  1% — ✓ 8 / ✗ 575
     ✗ Group-0-coingecko-OMG / ETH returned 200
      ↳  1% — ✓ 8 / ✗ 575
     ✗ Group-0-coingecko-AUTO / USD returned 200
      ↳  1% — ✓ 6 / ✗ 577
     ✗ Group-0-coingecko-DOT / USD returned 200
      ↳  0% — ✓ 2 / ✗ 581
     ✗ Group-0-coingecko-OCEAN / USD returned 200
      ↳  2% — ✓ 13 / ✗ 570
     ✗ Group-0-coingecko-RSR / USD returned 200
      ↳  0% — ✓ 2 / ✗ 581
     ✗ Group-0-coingecko-USDP / USD returned 200
      ↳  1% — ✓ 8 / ✗ 575
     ✗ Group-0-coingecko-APE / ETH returned 200
      ↳  0% — ✓ 4 / ✗ 579
     ✗ Group-0-coingecko-ACH / USD returned 200
      ↳  0% — ✓ 2 / ✗ 581
     ✗ Group-0-coingecko-ICP / USD returned 200
      ↳  0% — ✓ 3 / ✗ 580
     ✗ Group-0-coingecko-BICO / USD returned 200
      ↳  1% — ✓ 11 / ✗ 572
     ✗ Group-0-coingecko-SAND / USD returned 200
      ↳  1% — ✓ 8 / ✗ 575
     ✗ Group-0-coingecko-OGN / ETH returned 200
      ↳  1% — ✓ 7 / ✗ 576
     ✗ Group-0-coingecko-ENS / USD returned 200
      ↳  0% — ✓ 5 / ✗ 578
     ✗ Group-0-coingecko-DAI / ETH returned 200
      ↳  1% — ✓ 7 / ✗ 576
     ✗ Group-0-coingecko-AUDIO / USD returned 200
      ↳  0% — ✓ 5 / ✗ 578
     ✗ Group-0-coingecko-Total Marketcap / USD returned 200
      ↳  0% — ✓ 4 / ✗ 579
     ✗ Group-0-coingecko-1INCH / USD returned 200
      ↳  1% — ✓ 7 / ✗ 576
     ✗ Group-0-coingecko-SLP / ETH returned 200
      ↳  2% — ✓ 12 / ✗ 571
     ✗ Group-0-coingecko-TOMO / USD returned 200
      ↳  1% — ✓ 11 / ✗ 572
     ✗ Group-0-coingecko-LUSD / USD returned 200
      ↳  1% — ✓ 9 / ✗ 574
     ✗ Group-0-coingecko-GNO / ETH returned 200
      ↳  1% — ✓ 11 / ✗ 572
     ✗ Group-0-coingecko-XMR / USD returned 200
      ↳  2% — ✓ 13 / ✗ 570
     ✗ Group-0-coingecko-SHIB / ETH returned 200
      ↳  1% — ✓ 11 / ✗ 572
     ✗ Group-0-coingecko-CREAM / ETH returned 200
      ↳  0% — ✓ 5 / ✗ 578
     ✗ Group-0-coingecko-FXS / USD returned 200
      ↳  0% — ✓ 5 / ✗ 578
     ✗ Group-0-coingecko-ATOM / USD returned 200
      ↳  0% — ✓ 5 / ✗ 578
     ✗ Group-0-coingecko-RLC / ETH returned 200
      ↳  0% — ✓ 5 / ✗ 578
     ✗ Group-0-coingecko-AUCTION / USD returned 200
      ↳  1% — ✓ 6 / ✗ 577
     ✗ Group-0-coingecko-GHST / ETH returned 200
      ↳  2% — ✓ 15 / ✗ 568
     ✗ Group-0-coingecko-BADGER / ETH returned 200
      ↳  0% — ✓ 5 / ✗ 578

     checks.........................: 1.20%  ✓ 1734     ✗ 141684
     data_received..................: 62 MB  1.4 kB/s
     data_sent......................: 20 MB  459 B/s
     errors.........................: 98.79% ✓ 141684   ✗ 1734  
     http_req_blocked...............: avg=37.31µs min=90ns    med=198ns   max=5.09s    p(90)=347ns   p(95)=461ns   
     http_req_connecting............: avg=62ns    min=0s      med=0s      max=866.88µs p(90)=0s      p(95)=0s      
   ✗ http_req_duration..............: avg=1.75s   min=1.34ms  med=2.5s    max=5.51s    p(90)=2.5s    p(95)=2.51s   
       { expected_response:true }...: avg=1.47s   min=5.34ms  med=2.57s   max=3.63s    p(90)=2.75s   p(95)=2.85s   
   ✗ http_req_failed................: 98.79% ✓ 141684   ✗ 1734  
     http_req_receiving.............: avg=91.65µs min=16.75µs med=49.45µs max=56.2ms   p(90)=75.51µs p(95)=90.44µs 
     http_req_sending...............: avg=59.7µs  min=11.06µs med=51.11µs max=6.64ms   p(90)=83.67µs p(95)=111.06µs
     http_req_tls_handshaking.......: avg=1.29µs  min=0s      med=0s      max=78.63ms  p(90)=0s      p(95)=0s      
     http_req_waiting...............: avg=1.75s   min=1.23ms  med=2.5s    max=5.51s    p(90)=2.5s    p(95)=2.51s   
     http_reqs......................: 143418 3.319646/s
     iteration_duration.............: avg=1m14s   min=15.46s  med=1m13s   max=1m27s    p(90)=1m16s   p(95)=1m16s   
     iterations.....................: 583    0.013494/s
     vus............................: 1      min=1      max=1   
     vus_max........................: 1      min=1      max=1   `

**Review by austinborn — APPROVED** [review]
